docs/package_templates/header_only/all/conanfile.py

```diff
@@ -3,101 +3,72 @@
 
     def requirements(self):
         # Prefer self.requires method instead of requires attribute
         # Direct dependencies of header only libs are always transitive since they are included in public headers
-        self.requires("dependency/0.8.1", transitive_headers=True)
+        self.requires("dependency/0.8.1")
+        # Some dependencies on CCI are allowed to use version ranges.
+        # See https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges
+        self.requires("openssl/[>=1.1 <4]")
 
     # same package ID for any package
     def package_id(self):
         self.info.clear()
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            # Validate the minimum cpp standard supported when installing the package. For C++ projects only
-            check_min_cppstd(self, self._min_cppstd)
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(
-                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-            )
-
-        # In case this library does not work in some another configuration, it should be validated here too
+        # Validate the minimum cpp standard supported when installing the package. For C++ projects only
+        check_min_cppstd(self, 14)
+        # in case it does not work in another configuration, it should be validated here. Always comment the reason including the upstream issue.
+        # INFO: Upstream does not support DLL: See <URL>
         if self.settings.os == "Windows":
             raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
 
     def source(self):
         # Download source package and extract to source folder
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
-
-    # Not mandatory when there is no patch, but will suppress warning message about missing build() method
-    def build(self):
         # The attribute no_copy_source should not be used when applying patches in build
+        # Using patches is always the last resort to fix issues. If possible, try to fix the issue in the upstream project.
         apply_conandata_patches(self)
 
+    # Suppress warning message about missing build() method when running Conan
+    def build(self):
+        pass
+
     # Copy all files to the package folder
     def package(self):
         copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
-        copy(
-            self,
-            "*.h",
-            os.path.join(self.source_folder, "include"),
-            os.path.join(self.package_folder, "include"),
-        )
+        copy(self, "*.h", os.path.join(self.source_folder, "include"), os.path.join(self.package_folder, "include"))
```

Comment:
```suggestion
        # Prefers CMake.install() or similar in case the upstream offers an official method to install the headers.
        copy(self, "*.h", os.path.join(self.source_folder, "include"), os.path.join(self.package_folder, "include"))
```